src/Core/Foundation/Version/Version.php

```diff
@@ -49,37 +83,37 @@ public function __construct($version, $majorVersionString, $majorVersion, $minor
     }
 
     /**
-     * Returns the current PrestaShop version which is hardcoded in \AppKernel::VERSION
+     * Returns the current PrestaShop version which is hardcoded in \AppKernel::VERSION.
      *
      * @return string For example "1.7.4.0"
      */
-    public function getVersion()
+    public function getFullVersion()
     {
-        return $this->version;
+        return $this->fullVersion;
     }
 
     /**
-     * Returns the current PrestaShop major version as string or integer
+     * Returns the current PrestaShop major version as a string.
      *
-     * @param int $type Defines the datatype of the returned major version (\AppKernel::STRING or \AppKernel::INTEGER)
-     *
-     * @return int|string For example "1.7" or 17
+     * @return string For example "1.7"
      */
-    public function getMajorVersion($type = self::STRING)
+    public function getStringMajorVersion()
```

Comment:
Why 2 functions? Because I think a function shouldn't do 2 different things when we can do otherwise. More this make each function easier to test and easier to understand for your IDE (do I need a string or an integer?)